.docker/scripts/create-test-data.sh

```diff
@@ -12,20 +15,26 @@ fi
 
 add_user_if_needed () {
     echo "Adding user '$2' ($1) with email address [$3]"
-    ckan_cli user "$1" | grep "$1" || ckan_cli user add "$1"\
+    ckan_cli user show "$1" | grep "$1" || ckan_cli user add "$1"\
         fullname="$2"\
         email="$3"\
         password="${4:-Password123!}"
 }
 
-# We know the "admin" sysadmin account exists, so we'll use her API KEY to create further data
-API_KEY=$(ckan_cli user admin | tr -d '\n' | sed -r 's/^(.*)apikey=(\S*)(.*)/\2/')
+add_user_if_needed "${CKAN_USER_NAME}" "${CKAN_DISPLAY_NAME}" "${CKAN_USER_EMAIL}"
+ckan_cli sysadmin add "${CKAN_USER_NAME}"
+
+API_KEY=$(ckan_cli user show "${CKAN_USER_NAME}" | tr -d '\n' | sed -r 's/^(.*)apikey=(\S*)(.*)/\2/')
+if [ "$API_KEY" = "None" ]; then
+    echo "No API Key found on ${CKAN_USER_NAME}, generating API Token..."
+    API_KEY=$(ckan_cli user token add "${CKAN_USER_NAME}" test_setup |grep -v '^API Token created' | tr -d '[:space:]')
+fi
 
 # Creating test data hierarchy which creates organisations assigned to datasets
 ckan_cli create-test-data hierarchy
 
 # Creating basic test data which has datasets with resources
-ckan_cli create-test-data
+ckan_cli create-test-data basic
```

Comment:
is this 2.8 compatible?

Reply:
Yes, the tests are still passing on 2.8. It recognises "basic" as the default type, even though the paster docs don't explicitly say so.